pkg/arvo/sys/zuse.hoon

```diff
@@ -6512,8 +6512,13 @@
       ?^  t.rax
         [p.pok [ire q.pok]]:[pok=$(rax t.rax) ire=i.rax]
       =/  raf/(like term)
-          =>  |=(a/@ ((sand %tas) (crip (flop (trip a)))))
-          (;~(sfix (sear . sym) dot) [1^1 (flop (trip i.rax))])
+        %-  ;~  sfix
+              %+  sear
+                |=(a/@ ((sand %tas) (crip (flop (trip a)))))
```

Comment:
Does anyone know why the extension has to be a valid @tas here? This will still break on extensions like `.7z`.

Reply:
docs says "a term or @tas, an atomic ASCII string which obeys symbol rules: lowercase and digit only, infix hyphen, first character must be a lowercase letter." — not that the docs are up to date, but maybe this is a clue

Reply:
I don't think something like a term would change all that frequently really.

Reply:
The extension in `++pork` is defined as `(unit @ta)`, though, so I'm wondering why there's a coercion here to a more restrictive type.

Reply:
In general, the intention is to map file extensions to marks inside urbit. This generally makes us compatible with only a subset of the outside world.

A similar example is the use of atoms to encode arbitrary binary data. If we require precise compatibility with outside systems is required, as in the case of hash functions and other cryptographic use-cases, the possibility of trailing null bytes must be handled with a structure such as `$octs`. Similarly we have have base16 encoding a la `@ux`, and base16 encoding a la #3576.

Reply:
Hmm, not sure what to make of this. "File extensions as marks" is nice, but would prevent us from, well, implementing a world-compatible http client. Perhaps we should only start caring about this wherever url -> clay path conversions or similar things happen?